### PR TITLE
Add flux recipe for ci

### PR DESCRIPTION
**Pull request by yaoyu-33**
# What does this PR do ?

Add flux recipe for ci

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: nemo/collections/diffusion/recipes/flux_535m.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+import nemo_run as run
+import torch
+
+from megatron.core.distributed import DistributedDataParallelConfig
+from megatron.core.optimizer import OptimizerConfig
+
+from nemo import lightning as nl
+from nemo.collections import llm
+from nemo.collections.diffusion.models.flux.model import FluxModelParams, MegatronFluxModel
+from nemo.collections.llm.recipes.log.default import default_log, default_resume, tensorboard_logger
```
[the remark this answers is not in the document]
Comment (abhinavg4) [reply nemo/collections/diffusion/recipes/flux_535m.py:28]: I think this might have to be fixed

**Review by abhinavg4 — APPROVED** [review]
Look good

**Review by abhinavg4 — APPROVED** [review]
Looks good

**Review by Victor49152 — APPROVED** [review]
